buildscripts/kokoro/unix.sh

```diff
@@ -31,7 +31,7 @@ fi
 # ARCH is x86_64 unless otherwise specified.
 ARCH="${ARCH:-x86_64}"
 
-cat <<EOF >> gradle.properties
+cat <<'EOF' >> gradle.properties
```

Comment:
Note: this fixes broken syntax in some editors (including GitHub), as well as prevents from accidental parameter/arithmetic expansion, and command substitution:
![image](https://github.com/grpc/grpc-java/assets/672669/b5ef220e-8e70-4aa3-bb52-be8cca29060c)